### PR TITLE
James 1841

### PR DESCRIPTION
**Pull request by chibenwa**
Re-open of https://github.com/linagora/james-project/pull/443 targetting site branch

Up to date visuals : 

screenshot from 2016-10-21 14-47-31

screenshot from 2016-10-21 14-47-34

screenshot from 2016-10-21 14-47-37

screenshot from 2016-10-21 14-47-39

screenshot from 2016-10-21 14-47-46

screenshot from 2016-10-21 14-47-49

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

File: server/src/site/xdoc/config-mailetcontainer.xml [hunk 1]
```diff
@@ -27,17 +27,14 @@
...
-    <ul>
-      <li>No more mailetpackage specification in mailetcontainer.xml</li>
-      <li>Exactly one standard package, hard-coded in source code</li>
-      <li>Standard package applied to mailet classnames without package prefix</li>
-      <li>All custom mailets must be specified with full package prefix</li>
-      <li>Package prefixed custom mailet names will be searched across the entire applicable class path</li>
-      <li>Composite matchers</li>
-    </ul>
+      <p>This documents explains how to configure Mail processing upon dequeue. Mails pass threw the MailetContainer. The
+          MailetContainer is composed of Mailets (execution units that perform actions based on incoming mail) and of Matchers
```
Comment (mbaechler) [inline server/src/site/xdoc/config-mailetcontainer.xml]: it's not composed, it runs a Mailet / Matcher pipeline

File: server/src/site/xdoc/config-mailetcontainer.xml [hunk 2]
```diff
@@ -27,17 +27,14 @@
...
+          (condition for executing a mailet) arranged into processors (execution stream).</p>
```
Comment (mbaechler) [inline server/src/site/xdoc/config-mailetcontainer.xml]: what "processors" means in term of architecture ?

File: server/src/site/xdoc/config-mailetcontainer.xml [hunk 3]
```diff
@@ -27,17 +27,14 @@
...
-    <p>Changes between 3.0-M2 and 3.0-beta3:</p>
-    <ul>
-      <li>No more mailetpackage specification in mailetcontainer.xml</li>
-      <li>Exactly one standard package, hard-coded in source code</li>
-      <li>Standard package applied to mailet classnames without package prefix</li>
-      <li>All custom mailets must be specified with full package prefix</li>
-      <li>Package prefixed custom mailet names will be searched across the entire applicable class path</li>
-      <li>Composite matchers</li>
-    </ul>
+      <p>This documents explains how to configure Mail processing upon dequeue. Mails pass threw the MailetContainer. The
```
Comment (mbaechler) [inline server/src/site/xdoc/config-mailetcontainer.xml]: dequeue ?

File: server/src/site/xdoc/index.xml [hunk 4]
```diff
@@ -112,10 +112,10 @@
 	        <p>100% pure Java Mail server designed to be a complete and portable enterprise 
 	           mail engine solution.</p>
 	           
-	        <p>Supports currently available IETF protocols, including SMTP, LMTP, POP3 and IMAP4.</p>
+	        <p>Supports currently available IETF protocols, including SMTP, LMTP, POP3, IMAP4, ManageSIEVE. We also plan to support JMAP.</p>
 	         
 	        <p>Apache James Server is able to store user and message data either in a file-system (Maildir),
```
Comment (aduprat) [inline server/src/site/xdoc/index.xml]: I think JPA should come first, and Maildir last.

File: server/src/site/xdoc/index.xml [hunk 5]
```diff
@@ -56,9 +56,10 @@
 		</div>
       
 	    <p>The Apache Java Mail Server
-	      is a 100% pure Java SMTP, IMAP4 and POP3 Mail 
+	      is a 100% pure Java Mail
 	      server designed to be a complete and portable enterprise mail engine solution.  
-	      Apache James Server is based on currently available open protocols.</p>
+	      Apache James Server is based on currently available open protocols. IMAP, LMTP and SMTP is supported out of the box.
```
Comment (aduprat) [inline server/src/site/xdoc/index.xml]: ...LMTP and SMTP `are` supported...

File: server/src/site/xdoc/index.xml [hunk 6]
```diff
@@ -44,22 +44,14 @@
...
      <div id="tabs-1">
       
-	    <p>The Apache Java Mail Server
-	      is a 100% pure Java Mail
-	      server designed to be a complete and portable enterprise mail engine solution.  
-	      Apache James Server is based on currently available open protocols. IMAP, LMTP and SMTP is supported out of the box.
-          Support for JMAP, POP3 and ManageSieve is experimental.</p>
+	    <p>The server project wires together the different libraries composing James to provide a running service. It introduces
+            for example Netty wrappers for protocols, the Mail Queue component as well as Mailet container for mail processing,
+            and storage API for server related storage. Wiring is done threw Inversion of Control threw either Guice or Spring.</p>
```
Comment (aduprat) [inline server/src/site/xdoc/index.xml]: Wiring is done threw Inversion of Control `using` either ...

File: server/src/site/xdoc/config-mailetcontainer.xml (hunk 2 above)
Comment (chibenwa) [inline server/src/site/xdoc/config-mailetcontainer.xml]: List of mailet/matcher pairs allowing better logical organisation of MailetContainer

File: server/src/site/xdoc/index.xml (hunk 5 above)
Comment (chibenwa) [inline server/src/site/xdoc/index.xml]: Line was already modified

File: server/src/site/xdoc/config-mailetcontainer.xml [hunk 7]
```diff
@@ -27,9 +27,10 @@
 
   <section name="Mailet Container Configuration">
 
-      <p>This documents explains how to configure Mail processing upon dequeue. Mails pass threw the MailetContainer. The
-          MailetContainer is composed of Mailets (execution units that perform actions based on incoming mail) and of Matchers
-          (condition for executing a mailet) arranged into processors (execution stream).</p>
+      <p>This documents explains how to configure Mail processing. Mails pass through the MailetContainer. The
+          MailetContainer is a Mailets (execution units that perform actions based on incoming mail) and Matchers
```
Comment (mbaechler) [inline server/src/site/xdoc/config-mailetcontainer.xml]: It's more logical to start with Matchers, don't you think ?

**Pull request merged**